src/solution.rs

```diff
@@ -1,45 +1,45 @@
 use std::fmt;
 use std::rc::Rc;
+use std::sync::Arc;
 
 use crate::variable::Variable;
 use crate::{ffi, scip_call_panic};
+use crate::scip::ScipPtr;
 
 /// A wrapper for a SCIP solution.
-#[derive(PartialEq, Eq)]
-pub struct Solution<'a> {
-    pub(crate) scip_ptr: *mut ffi::SCIP,
+pub struct Solution {
+    pub(crate) scip_ptr: Arc<ScipPtr>,
     pub(crate) raw: *mut ffi::SCIP_SOL,
```

Comment:
I think it might be good for the future to have this eventually a `NotNull`. But it is outside this pull's scope

Reply:
I added issue #157 to track this